entity/c/c_crypto.h

```diff
@@ -0,0 +1,181 @@
+#ifndef C_CRYPTO_H
+#define C_CRYPTO_H
+
+#include <openssl/aes.h>
+#include <openssl/bio.h>
+#include <openssl/err.h>
+#include <openssl/evp.h>
+#include <openssl/hmac.h>
+#include <openssl/pem.h>
+#include <openssl/rand.h>
+#include <openssl/rsa.h>
+#include <openssl/sha.h>
+#include <openssl/ssl.h>
+#include <openssl/x509.h>
+
+#include "c_common.h"
+
+#define AES_CBC_128_KEY_SIZE 128
+#define AES_CBC_128_IV_SIZE 16
+#define DIST_KEY_EXPIRATION_TIME_SIZE 6
+#define KEY_EXPIRATION_TIME_SIZE 6
+#define SESSION_KEY_ID_SIZE 8
+#define ABS_VALIDITY_SIZE 6
+#define REL_VALIDITY_SIZE 6
+#define MAC_KEY_SIZE 32
+#define CIPHER_KEY_SIZE 16
+#define RSA_KEY_SIZE 256
+#define RSA_ENCRYPT_SIGN_SIZE RSA_KEY_SIZE * 2
+#define IV_SIZE 16
+#define SHA256_DIGEST_LENGTH 32
+
+// Struct for digital signature
+typedef struct {
+    unsigned char data[RSA_KEY_SIZE];
+    unsigned char sign[RSA_KEY_SIZE];
+} signed_data_t;
+
+// Must free mac key & cipher key
```

Comment:
Ideally, we should provide `free` functions for the structs, like OpenSSL does: https://www.openssl.org/docs/man3.0/man3/OPENSSL_free.html

For example, `void free_distribution_key(distribution_key_t* dist_key);`

Reply:
Done.
`void free_SST_ctx(SST_ctx_t *ctx)` and `void free_session_key_list_t(session_key_list_t *session_key_list)` added.